lib/shopify_cli/git.rb

```diff
@@ -46,12 +46,75 @@ def sha(dir: Dir.pwd, ctx: Context.new)
       end
 
       ##
-      # will make calls to git to clone a new repo into a supplied destination,
-      # it will also output progress of the cloning process.
+      # returns array with components of git clone command
       #
       # #### Parameters
       #
-      # * `repository` - a git url for git to clone the repo from
+      # * `repo` - repo url without branch name
+      # * `dest` - a filepath to where the repo should be cloned to
+      # * `branch` - branch name when cloning
+      #
+      # #### Returns
+      #
+      # * array of strings
+      #
+      # #### Example
+      #
+      #   ["clone", "--single-branch", "--branch", "test-branch", "test-app"]
+      #
+      def git_clone_command(repo, dest, branch)
+        if branch
+          ["clone", "--single-branch", "--branch", branch, repo, dest]
+        else
+          ["clone", "--single-branch", repo, dest]
+        end
+      end
+
+      ##
+      # calls git to clone a new repo into a supplied destination,
+      # it will also call a supplied block with the percentage of clone completion
+      #
+      # #### Parameters
+      #
+      # * `repo_with_branch` - a git url for git to clone the repo from
+      # * `dest` - a filepath to where the repo should be cloned to
+      # * `ctx` - the current running context of your command, defaults to a new context.
+      #
+      # #### Returns
+      #
+      # * `sha_string` - string of the sha of the most recent commit to the repo
+      #
+      # #### Example
+      #
+      #   ShopifyCLI::Git.raw_clone('git@github.com:shopify/test.git', 'test-app')
+      #
+      def raw_clone(repo_with_branch, dest, ctx: Context.new)
+        if Dir.exist?(dest) && !Dir.empty?(dest)
+          ctx.abort(ctx.message("core.git.error.directory_exists"))
+        else
+          msg = []
+          require "open3"
```

Comment:
I don't think that if `raw_clone` and `clone` are both called `open3` will be required twice?